### ISSUE TITLE
how to make the cpu 100% usage?

### ISSUE THREAD

**Issue by viponedream** [gensyn-ai/rl-swarm#257]
now it's only 50% usage.
![Image](https://github.com/user-attachments/assets/304a67e5-0e73-41e5-832a-905fd699db6e)

**Comment by DonS598** [comment]
Change /root/rl-swarm/hivemind_exp/configs/mac/grpo-qwen-2.5-0.5b-deepseek-r1.yaml   file config this way bro

# Model arguments
model_revision: main
torch_dtype: bfloat16
bf16: false
tf32: false

**Comment by viponedream** [comment]
> Change /root/rl-swarm/hivemind_exp/configs/mac/grpo-qwen-2.5-0.5b-deepseek-r1.yaml file config this way bro
> 
> # Model arguments
> model_revision: main torch_dtype: bfloat16 bf16: false tf32: false

thanks for your reply.
but it's the same result.
my vps 16h32G,
it always run in  8vcores even i change the config you gave me.
can you post the full config content?

**Comment by DonS598** [comment]
This is content. But I didn't change anything exept what I wrote before

# Model arguments
model_revision: main
torch_dtype: bfloat16
bf16: false
tf32: false

# Dataset arguments
dataset_id_or_path: 'openai/gsm8k'

# Training arguments
max_steps: 20 # Original 450
gradient_accumulation_steps: 4
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
learning_rate: 5.0e-7 # 1.0e-6 as in the deepseek math paper 5-e7 from https://hijkzzz.notion.site/unraveling-rlhf-and-its-variants-engineering-insights#147d9a33ecc9806090f3d5c749d31f05
lr_scheduler_type: cosine
warmup_ratio: 0.03

# GRPO arguments
use_vllm: false
num_generations: 2
per_device_train_batch_size: 2
beta: 0.001 # 0.04 as in the deepseek math paper 0.001 from https://hijkzzz.notion.site/unraveling-rlhf-and-its-variants-engineering-insights#147d9a33ecc9806090f3d5c749d31f05
max_prompt_length: 256
max_completion_length: 1024

# Logging arguments
logging_strategy: steps
logging_steps: 2
report_to:
- tensorboard
save_strategy: "steps"
save_steps: 25
seed: 42

# Script arguments
max_rounds: 10000

# Model-specific arguments
model_name_or_path: unsloth/Qwen2.5-0.5B-Instruct
output_dir: runs/gsm8k/multinode/Qwen2.5-0.5B-Instruct-Gensyn-Swarm
